src/common/m_variables_conversion.fpp

```diff
@@ -1518,8 +1520,23 @@ contains
                     end if
 
                     ! Computing the energy from the pressure
-                    E_K = gamma_K*pres_K + pi_inf_K &
-                          + 5e-1_wp*rho_K*vel_K_sum + qv_K
+
+                    if (chemistry) then
+                        !$acc loop seq
+                        do i = chemxb, chemxe
+                            Y_K(i - chemxb + 1) = qK_prim_vf(j, k, l, i)
+                        end do
+                        !Computing the energy from the internal energy of the mixture
+                        call get_mixture_molecular_weight(Y_k, mix_mol_weight)
+                        R_gas = gas_constant/mix_mol_weight
+                        T_K = pres_K/rho_K/R_gas
+                        call get_mixture_energy_mass(T_K, Y_K, E_K)
+                        E_K = rho_K*E_K + 5e-1_wp*rho_K*vel_K_sum
+                    else
+                        ! Computing the energy from the pressure
+                        E_K = gamma_K*pres_K + pi_inf_K &
+                              + 5e-1_wp*rho_K*vel_K_sum + qv_K
+                    end if
```

Comment:
Can you make sure this doesn't have any performance impact on non-chemistry cases? This variables conversion subroutine can be really touchy on AMD hardware...